src/main/java/edu/harvard/iq/dataverse/util/json/JSONLDUtil.java

```diff
@@ -80,19 +80,14 @@ public static Dataset updateDatasetMDFromJsonLD(Dataset ds, String jsonLDBody,
         DatasetVersion dsv = new DatasetVersion();
 
         JsonObject jsonld = decontextualizeJsonLD(jsonLDBody);
-        String id = null;
-        try {
-            id=jsonld.getString("@id");
-        } catch (NullPointerException npe) {
-            //Do nothing - a null value and other invalid values will be caught in parsing
-        }
-        Optional<GlobalId> maybePid = GlobalId.parse(id);
-        if (maybePid.isPresent()) {
-            ds.setGlobalId(maybePid.get());
-        } else {
-            if(migrating) {
-              // unparsable PID passed. Terminate.
-              throw new BadRequestException("Cannot parse the @id. Make sure it is in valid form - see Dataverse Native API documentation.");
+        if (migrating) {

```

Comment:
FWIW: Submitting an '@id' value for the dataset PID, and a modificationdate (next diff) are only allowed for migrate (not valid in the create dataset case), so these are cleaner fixes to just skip running into the null values in the create case.